raiden/tests/utils/transfer.py

```diff
@@ -364,28 +378,253 @@ def transfer_and_assert_path(
+        )
+
+        if node1_deposit_event is not None:
+            msg = "Node1 has fetched and ignored the node0's deposits, this is likely a bug."
+        elif node0_deposit_event.deposit_transaction.deposit_block_number >= block_number1:
+            msg = (
+                "Node1's has a problem with its blockchain event filters, it "
+                "missed node0's deposit event even though it has seen a newer "
+                "confirmed block"
+            )
+        elif not app0.raiden.alarm:
+            msg = (
+                "Node1 has not seen the block at which node0's deposit happened "
+                "and the alarm task is not running. Either the test stopped "
+                "the node before it had time or the node got killed because of "
+                "another error."
+            )
+        else:
+            msg = (
+                "Node1 has not yet seen the block at which node0's deposit "
+                "happened. The test is likely missing synchronization."
+            )
+
+        msg = (
+            f"{msg}. "
+            f"node1={to_checksum_address(app1.raiden.address)} "
+            f"node0={to_checksum_address(app0.raiden.address)} "
+            f"block_number0={block_number0} "
+            f"block_number1={block_number1} "
+            f"state_change_id0={saved_state0.state_change_id} "
+            f"state_change_id1={saved_state1.state_change_id}."
+        )
+
+        raise AssertionError(msg)
+
+
+def assert_balance_proof(
+    token_network_address: TokenNetworkAddress,
+    app0: App,
+    app1: App,
+    saved_state0: SavedState,
+    saved_state1: SavedState,
+) -> None:
+    """Assert app0 and app1 agree on the latest balance proof from app0.
+
+    Notes:
+        - The other direction of the channel does not have to be synchronized,
+          it can be checked with another call.
+        - It is important to do the validation on a fixed  state, that is why
+          saved_state0 is used.
+    """
+    assert app0.raiden.address == saved_state0.state.our_address
+    assert app1.raiden.address == saved_state1.state.our_address
+
+    channel0 = views.get_channelstate_by_token_network_and_partner(
+        saved_state0.state, token_network_address, app1.raiden.address
+    )
+    channel1 = views.get_channelstate_by_token_network_and_partner(
+        saved_state1.state, token_network_address, app0.raiden.address
+    )
+
+    balanceproof0: BalanceProofUnsignedState = channel0.our_state.balance_proof
+    balanceproof1: BalanceProofSignedState = channel1.partner_state.balance_proof
+
+    if balanceproof0 is None:
+        msg = "Bug detected. The sender does not have a balance proof, but the recipient does."
+        assert balanceproof1 is None, msg
+
+        # nothing to compare
+        return
+
+    # Handle the case when the recipient didn't receive the message yet.
+    if balanceproof1 is not None:
```

Comment:
general style question: do we avoid `var = if A else B` conditional assignments on the code base?

Reply:
I generally do, however this is not part of our style guide.